Community/Tdarr_Plugin_dt0y_DaveThe0nly_HWACCEL_SINGLE_TRANSCODER.js

```diff
@@ -0,0 +1,576 @@
+      },
+      tooltip: 'Sets quality of x264 being preset:crf (veryfast:27) being default',
+    },
+    {
+      name: 'createOptimizedAudioTrack',
+      type: 'string',
+      defaultValue: 'ac3:6,aac:6,aac:3',
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: 'Creates an optimized audio track/s codec:channels (aac:2,ac3:6)',
+    },
+    {
+      name: 'keepAudioTracks',
+      type: 'string',
+      defaultValue: 'eng',
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: 'Keeps an audio track of selected language/s, separated by ",", if empty all are kept',
+    },
+    {
+      name: 'outputContainer',
+      type: 'string',
+      defaultValue: '.mp4',
+      inputUI: {
+        type: 'dropdown',
+        options: ['.mp4', '.mkv', 'keep'],
+      },
+      tooltip: 'Container',
+    },
+    {
+      name: 'extractSubtitles',
+      type: 'string',
+      defaultValue: 'none',
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: 'Extracts subtitles by language from the file',
+    },
+    {
+      name: 'subtitleFormat',
+      type: 'string',
+      defaultValue: 'srt,ass', // most compatible with plex
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: 'Comma separated list of subtitle formats to create',
+    },
+  ],
+});
+
+const plugin = (file, librarySettings, inputs, otherArguments) => {
+  const lib = require('../methods/lib')();
+
+  const {
+    targetResolutionBitrate: _targetBitrate,
+    extractSubtitles: _extractSubtitles,
+    outputContainer,
+    hwAccelerate,
+    forceHevc,
+    createOptimizedAudioTrack: _createOptimizedAudioTrack,
+    keepAudioTracks: _keepAudioTracks,
+    hevcCompressionLevel,
+    x264quality: _x264quality,
+    subtitleFormat: _subtitleFormat,
+  } = lib.loadDefaultValues(inputs, details);
+
+  const resolution = file.video_resolution === 'DCI4K' ? '4KUHD' : file.video_resolution;
+
+  // If target bitrate is set, try and get it from the setting otherwise set to Infinity to ignore the resolution
+  const targetBitrate = _targetBitrate
+    ? createBitRateFromString(_targetBitrate.split(','))[resolution] || Infinity
+    : Infinity;
+  const keepAudioTracks = _keepAudioTracks.split(',');
+  const extractSubtitles = _extractSubtitles.split(',');
+  const subtitleFormat = _subtitleFormat.split(',');
+
+  const response = {
+    processFile: false,
+    preset: '',
+    container: outputContainer === 'keep' ? `.${file.container}` : outputContainer,
+    handBrakeMode: false,
+    FFmpegMode: true,
+    infoLog: '',
+    file,
+    removeFromDB: false,
+    updateDB: false,
+    reQueueAfter: false,
+  };
+
+  if (file.fileMedium !== 'video') {
+    response.processFile = false;
+    response.infoLog += 'File is not a video. Exiting \n';
+    return response;
+  }
+
+  if (!file.ffProbeData.streams.length) {
+    response.processFile = false;
+    response.infoLog += 'No streams to process!! \n';
+    return response;
+  }
```

Comment:
In these cases the file should probably just error out instead of silently fail like it does now.